agenta-web/src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,244 @@
+import React, {useState} from "react"
+import {Filter, JSSTheme} from "@/lib/Types"
+import {ArrowCounterClockwise, CaretDown, Funnel, Plus, Trash, X} from "@phosphor-icons/react"
+import {Button, Divider, Input, Popover, Select, Space, Typography} from "antd"
+import {createUseStyles} from "react-jss"
+import {useUpdateEffect} from "usehooks-ts"
+
+const useStyles = createUseStyles((theme: JSSTheme) => ({
+    popover: {
+        "& .ant-popover-inner": {
+            width: "600px !important",
+            padding: `0px ${theme.paddingXS}px ${theme.paddingXS}px ${theme.padding}px`,
+        },
+    },
+    filterHeading: {
+        fontSize: theme.fontSizeHeading5,
+        lineHeight: theme.lineHeightHeading5,
+        fontWeight: theme.fontWeightMedium,
+    },
+    filterContainer: {
+        padding: 8,
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "start",
+        borderRadius: theme.borderRadius,
+        backgroundColor: "#f5f7fa",
+        marginTop: 8,
+    },
+}))
+
+type Props = {
+    filterData?: Filter[]
+    columns: {value: string; label: string}[]
+    onApplyFilter: (filters: Filter[]) => void
+    onClearFilter: (filters: Filter[]) => void
+}
+
+const Filters: React.FC<Props> = ({filterData, columns, onApplyFilter, onClearFilter}) => {
+    const classes = useStyles()
+    const emptyFilter = [{key: "", operator: "", value: ""}] as Filter[]
+
+    const [filter, setFilter] = useState<Filter[]>(emptyFilter)
+    const [isFilterOpen, setIsFilterOpen] = useState(false)
+
+    useUpdateEffect(() => {
+        if (filterData && filterData.length > 0) {
+            setFilter(filterData)
+        } else {
+            setFilter(emptyFilter)
+        }
+    }, [filterData])
+
+    const operators = [
```

Comment:
missing `is` and `is_not` -- @ashrafchowdury

Reply:
missing `in` for lists and missing lists altogether -- @ashrafchowdury 
this may be for later.